feature/extract multiple numbers from text

JarbasAl opened the pull request:
## Description

extract_number assumes only 1 number is present in text, this adds a extract_numbers method to support cases like "test one two three"

normalize only accounts for numbers up to 20 (word to digit), method is improved to handle all extracted numbers by above method (needs improvements in further PRs)

## How to test

check unit tests

## Contributor license agreement signed?
CLA [ yes] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

pep8speaks commented:
Hello @JarbasAl! Thanks for submitting the PR.

 - There are no PEP8 issues in the file [`mycroft/util/lang/parse_en.py`](https://github.com/JarbasAl/mycroft-core/blob/82b4ec6482340fab9ca68c66f8b51fc12d170ec6/mycroft/util/lang/parse_en.py) !

 - There are no PEP8 issues in the file [`mycroft/util/parse.py`](https://github.com/JarbasAl/mycroft-core/blob/82b4ec6482340fab9ca68c66f8b51fc12d170ec6/mycroft/util/parse.py) !

 - There are no PEP8 issues in the file [`test/unittests/util/test_parse.py`](https://github.com/JarbasAl/mycroft-core/blob/82b4ec6482340fab9ca68c66f8b51fc12d170ec6/test/unittests/util/test_parse.py) !

forslund commented on mycroft/util/lang/parse_en.py:
```diff
@@ -1039,15 +1040,21 @@ def normalize_en(text, remove_articles):
         normalized += " " + word
 
+    # needs to be here to avoid import error
+    from mycroft.util.parse import extract_numbers
```
I'm not a fan of doing these kinds of workarounds for cyclic imports.

What do you think of creating a base method for extract_numbers in `lang/parse_common.py` which takes the extract_number function as an argument and a just a thin wrapper function in `util/parse.py`?

JarbasAl replied:
sounds good!

JarbasAl commented:
added support for more languages, but since i don't speak them i would appreciate someone to write unittests

it is using the "extract number" logic already implemented for each language recursively, so it should be correct

forslund commented:
Is the intent to only match numbers directly following eachother?

"so two beers for two bears" should return `[2]` and not `[2, 2]`?

JarbasAl commented:
it should extract every number, "1 dog, seven pigs, macdonald had a farm, 3 times 5 macarena" should also work

forslund commented:
The example I gave above didn't work for me.

JarbasAl commented:
you are right, i need to account for duplicate numbers, what is happening is that when it find the first "two" it replaces all instances in the string instead of only the consumed one

will account for this

JarbasAl commented:
moved main logic into parse_common to make it easier for supporting other languages

improved normalization of numbers, now "two twenty two" correctly becomes "2 22" instead of "2 twenty 2"

JarbasAl commented:
geez, how come pycharms auto format is creating new pep8 issues instead of solving them...

forslund commented:
Looks good. I can't figure out how to clean up the commit history so I'll just squash it all into one commit and merge it.

forslund commented:
Ok a conflict seem to have occurred when I merged the datetime update. Can you look at it?

JarbasAl commented:
rebased and added some more relevant unittests, wanted to leave some TODOs in there to make some limitations clear

follow up PR for pronounce_numbers (ordinals and long scale support) should fix some of those

follow up PR for issue https://github.com/MycroftAI/mycroft-core/issues/1875 will fix remaining edge cases
